File: src/main/java/top/dsbbs2/bukkitcord/utils/command/CommandUtil.java
package top.dsbbs2.bukkitcord.utils.command;

import java.util.*;

public class CommandUtil {
    public static Vector<String> commandListToCommandNameList(Vector<IChildCommand> v) {
        Vector<String> ret = new Vector<String>();
        for(i : v) {
            ret.add(i.class.getSimpleName());
        }
        return ret;
    }

    public static IChildCommand getCommand(Vector<IChildCommand> l, String cmd) {
        for(i : l) {
            if (i.class.getSimpleName().equalsIgnoreCase(cmd)) return i;
        }
        return null;
    }
}

File: src/main/java/top/dsbbs2/bukkitcord/utils/command/CommandUtil.kt
package top.dsbbs2.bukkitcord.utils.command

import top.dsbbs2.bukkitcord.utils.command.IChildCommand
import java.util.*

object CommandUtil {
    fun commandListToCommandNameList(v: Vector<IChildCommand>): Vector<String> {
        val ret = Vector<String>()
        for (i in v) {
            ret.add(i.javaClass.simpleName)
        }
        return ret
    }

    fun getCommand(l: Vector<IChildCommand>, cmd: String?): IChildCommand? {
        for (i in l) {
            if (i.javaClass.simpleName.equals(cmd, ignoreCase = true)) return i
        }
        return null
    }
}